### PR TITLE
Feature/m0081 meja 1662 base changelog

### PR DESCRIPTION
**Pull request by frivieremicroej**

### CODE REVIEW

File: VEEPortingGuide/architecture7Migration.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
+.. _architecture7_migration:
+
+Architecture 7.x Migration
+==========================
+
+This chapter describes the steps to migrate a VEE Port from Architecture ``7.x`` to Architecture ``8.0.0``.
+
+As a reminder, refer to the :ref:`Architecture 8.0.0 Changelog <changelog-8.0.0>` section for getting the full list of changes and updates.
```
Comment (jerome-leroux) [inline VEEPortingGuide/architecture7Migration.rst:8]:
```suggestion
As a reminder, refer to the :ref:`Architecture 8.0.0 Changelog <changelog-8.0.0>` section for the complete list of changes and updates.
```

File: VEEPortingGuide/architecture7Migration.rst [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+Update Platform Configuration Additions
+---------------------------------------
+
+Architecture ``8.0.0`` now directly integrates the :ref:`bsp_connection` mechanism.
+The Platform Configuration Additions must be updated accordingly. 
+Contact `our support team <https://www.microej.com/contact/#form_2>`_ to get the detailed instructions.
```
Comment (jerome-leroux) [inline VEEPortingGuide/architecture7Migration.rst:15]:
```suggestion
Contact `our support team <https://www.microej.com/contact/#form_2>`_ to get detailed instructions. 
```

File: VEEPortingGuide/architecture7Migration.rst [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+---------------------------------------
+
+Architecture ``8.0.0`` now directly integrates the :ref:`bsp_connection` mechanism.
+The Platform Configuration Additions must be updated accordingly. 
+Contact `our support team <https://www.microej.com/contact/#form_2>`_ to get the detailed instructions. 
+
+Update BSP with new Sections Names
+----------------------------------
+
+The Core Engine sections have been renamed to respect the standard ELF convention. See :ref:`Core Engine Link <core_engine_link>` section.
```
Comment (jerome-leroux) [inline VEEPortingGuide/architecture7Migration.rst:20]:
```suggestion
The Core Engine sections have been renamed to respect the standard ELF convention. See :ref:`Core Engine Link <core_engine_link>` section for further details.
```

File: VEEPortingGuide/images/Example-Architecture8.ld [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  {
+    . = ALIGN(4);
+    *(.bss)
+    *(.bss*)
+    *(COMMON)
+    
+    *(.bss.microej.stacks)
+    *(.bss.microej.runtime)
+    *(.bss.microej.heap)
+    *(.bss.microej.immortals)
```
Comment (jerome-leroux) [inline VEEPortingGuide/images/Example-Architecture8.ld:44]: Add `.bss.microej.features`?

File: VEEPortingGuide/architecture7Migration.rst [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+
+Migrate Device Module
+~~~~~~~~~~~~~~~~~~~~~
+
+The latest `Device Pack`_ available on the :ref:`central_repository` is backward compatible with the built-in Architecture module.
+
+.. _Device Pack: https://repository.microej.com/modules/com/microej/pack/device/device-pack/
+
+The following dependency must be added the :ref:`module.ivy <mmm_module_description>` of the VEE Port configuration project:
```
Comment (jerome-leroux) [inline VEEPortingGuide/architecture7Migration.rst:62]:
```suggestion
The following dependency must be added to the :ref:`module.ivy <mmm_module_description>` of the VEE Port configuration project:
```

File: VEEPortingGuide/architecture7Migration.rst [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+      
+     <dependency org="com.microej.pack.device" name="device-pack" rev="1.1.1" />
+
+Migrate ECOM-COMM Module
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are two options:
+  
+- either migrate to the latest ECOM-COMM Pack,
+- or integrate the legacy ECOM-COMM Pack files as-is into your VEE Port ``dropins`` folder,
```
Comment (jerome-leroux) [inline VEEPortingGuide/architecture7Migration.rst:74]:
```suggestion
- or integrate the legacy ECOM-COMM Pack files as-is into your VEE Port ``dropins`` folder.
```

File: VEEPortingGuide/architectureChangelog.rst [hunk 7]
```diff
@@ -33,6 +33,51 @@ specific configuration:
...
+    See :ref:`feature_portability_control` for more details.
+
+If you plan to migrate a VEE Port from Architecture ``7.x`` to Architecture ``8.x``, consider the :ref:`architecture7_migration` chapter.
+
+Core Engine
+~~~~~~~~~~~
+
+- Renamed :ref:`Core Engine sections <core_engine_link>` to fully respect the ELF standard naming convention. 
+- Removed check when passing a non-immortal array in SNI if VEE Port option ``core.sni.nonimmortal.access`` was set to ``false``.
+- Removed `LLBSP_isInReadOnlyMemory` in Core Engine Abstraction Layer (`LLBSP.h` file).
```
Comment (jerome-leroux) [inline VEEPortingGuide/architectureChangelog.rst:62]: Should be added to the migration guide?

Comment (jerome-leroux) [inline VEEPortingGuide/architectureChangelog.rst:62]:
```suggestion
- Removed ``LLBSP_isInReadOnlyMemory`` in Core Engine Abstraction Layer (``LLBSP.h`` file).
```

**Review by jerome-leroux — CHANGES_REQUESTED** [review]

File: VEEPortingGuide/images/Example-Architecture8.ld (hunk 4 above)
Comment (frivieremicroej) [reply VEEPortingGuide/images/Example-Architecture8.ld:44]: done

File: VEEPortingGuide/images/Example-Architecture8.ld [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  } >FLASH
+
+  .rodata_ro :
+  {
+    . = ALIGN(4);
+
+    *(.rodata)         /* .rodata sections (constants, strings, etc.) */
+    *(.rodata*)        /* .rodata* sections (constants, strings, etc.) */
+    *(.rodata.microej.soar.*)
+    *(.rodata.microej.resource.*)
```
[the remark this answers is not in the document]
Comment (frivieremicroej) [reply VEEPortingGuide/images/Example-Architecture8.ld:28]: done. fixed .rodata.microej.soar (single section)

File: VEEPortingGuide/architectureChangelog.rst (hunk 7 above)
Comment (frivieremicroej) [reply VEEPortingGuide/architectureChangelog.rst]: done

File: VEEPortingGuide/images/Example-Architecture8.ld (hunk 4 above)
Comment (frivieremicroej) [reply VEEPortingGuide/images/Example-Architecture8.ld:41]: also, the screenshot was not in sync with the shared .ld files

**Review by jerome-leroux — APPROVED** [review]

**Pull request merged**